ursa.ino

```diff
@@ -20,6 +20,22 @@ rmt_config_t configL;//settings for RMT pulse for stepper motor
 rmt_item32_t itemsL[1];//holds definition of pulse for stepper motor
 rmt_config_t configR;
 rmt_item32_t itemsR[1];
+byte voltage = 0; //0v=0 13v=255
+int signalStrength = -100;
+byte wifI = 0; //counter for wifi sending and recieving
+volatile byte recvdData[50] = {0}; //array to hold data recieved from DS.
+volatile byte recvdDataSize = 0; //how many bytes of the array were actually filled with a message from the DS
+volatile boolean recvdNewData = true;//set true when data gotten, set false when parsed
+volatile byte tosendData[50] = {0}; //array to hold data to send to DS.
+volatile byte tosendDataSize = 0; //how many bytes of the array are actually being used and need to be sent
+//since multiple tasks are running at once, we don't want two tasks to try and use one array at the same time.
+SemaphoreHandle_t mutexRecv;//used to check whether receiving tasks can safely change shared variables
+SemaphoreHandle_t mutexSend;//used to check whether sending tasks can safely change shared variables
+byte numAuxRecv = 0; //how many bytes of control data for extra things
+byte auxRecvArray[12] = {0}; //size of numAuxRecv
+byte numSendAux = 0; //how many bytes of sensor data to send
+byte auxSendArray[12] = {0}; //size of numAuxSend
+boolean settings = false; //will new PID settings be sent next

```

Comment:
This variable would likely be better expressed as `sendSettings` or `shouldSendSettings`

Reply:
@andrew it's really setSettings, or parseSettings, it is used so the DS can tell the robot whether it is about to send new PID settings for the robot to change-saving how much data needs to be sent for most packets.

Reply:
That works too